net/NetUtil.cpp

```diff
@@ -129,23 +96,46 @@ HostEntry::HostEntry(const std::string& desc, const char* port)
     , _errno(0)
     , _eaino(0)
 {
-    initFromHostName(desc, port);
-}
+    addrinfo hints;
+    std::memset(&hints, 0, sizeof(hints));
+    hints.ai_flags = AI_CANONNAME | AI_ADDRCONFIG;
 
-HostEntry::~HostEntry()
-{
+    addrinfo* ainfo = nullptr;
+    int rc = getaddrinfo(desc.c_str(), port, &hints, &ainfo);
```

Comment:
::getaddrinfo(..) ? (global namespace)

Reply:
personally I force myself to use ::system_call(..) to hint to me it is such a global function and not a method by any chance. Just taste.

Reply:
yeah, reasonable